scalableminds/webknossos
# dump & diff postgres schema

### TODO:
* [x] make folder customizable to allow diffs
* [x] add script for diffs (should allow for an actual db or a bunch of sql scripts)
* [x] this should not take into account the line order
* [x] add CI check that `tools/postgres/schema.sql` and `conf/evolutions/*` agree (fixes #2556)
* [x] check that no evolution number is used twice
* [x] produce a warning/error on startup, if schema does not fit (fixes #2655, also useful in production)
* [x] Bug: triggers and functions seem to be missing in dump

#### after deployment:
* [ ] run evolution `014`, make sure that the production instances agree with schema.sql

### Steps to test:
- check output on local startup with an up-to-date and an outdated schema
- check that the e2e tests fail, if the evolutions and schema.sql don't agree (see CI build before commit `equalize schema and evolutions`)
- double check in code review that the production databases can not be altered or dropped by those changes
- test evolution `014`

### Issues:
- fixes #2556 and #2655
------
